### ISSUE TITLE
🌙 Does not support animated avatars

### ISSUE THREAD

**Issue by Swarwerth** [Mr-KayJayDee/discord-image-generation#10]
Hello!
I tried using the module with an animated avatar (of a Nitro user), but here's what happens:

![Zerio js_triggered](https://user-images.githubusercontent.com/26115962/103277563-a32fb900-49c9-11eb-9595-8597911dba0e.gif)
![Zerio js_blur](https://user-images.githubusercontent.com/26115962/103277568-a5921300-49c9-11eb-9107-83bfc1cf153a.gif)

Here's my code:
```js
const user = message.mentions.users.first() || message.author;

await message.channel.send(`🎭 Génération de l'image...`);

const img = await new DIG.Jail().getImage(user.displayAvatarURL({dynamic: true, size: 4096, format: 'png'}));
const attach = new MessageAttachment(img, `${user.username}_blur.gif`);

const errorMentionEmbed = new MessageEmbed()
  .setColor('#c43131')
  .setAuthor(`💢 Erreur !`)
  .addField(`Je n'ai pas pu bien exécuter la commande \`jail\` !`, `Merci de mentionner un utilisateur du serveur !`, false)
  .setFooter(message.author.tag, message.author.displayAvatarURL({dynamic: true, size: 4096, format: 'png'}))
  .setTimestamp();

if (!args[0]) return message.channel.send(attach);
else return args[0] = message.mentions.users.first() ? message.channel.send(attach) : message.channel.send(errorMentionEmbed);
```

For the moment, I set `user.displayAvatarURL({dynamic: false)` to fix this bug.

**Comment by Mr-KayJayDee** [comment]
Hello, this is a known issue, but at the moment I'm looking for a way to add it, the next update will surely be that. 

Thanks for your patience. 🙂

**Comment by Swarwerth** [comment]
> Hello, this is a known issue, but at the moment I'm looking for a way to add it, the next update will surely be that.
> 
> Thanks for your patience. 🙂

Ok thanks! 😄

**Comment by Mr-KayJayDee** [comment]
Edit: I'm pausing the search for this feature because I can't manage to get all frame correctly. 

Plus, the generation time will be much higher, from 2~3 seconds to 15~20 seconds after my tests. 

Maybe I need to add something that allow the user to choose whether he wants to supports gif or not.

**Comment by asdbee** [comment]
This is fixed with a simple .replace('.gif','.png'). I tested .replace method and it look to fix this.

**Comment by Swarwerth** [comment]
> This is fixed with a simple .replace('.gif','.png'). I tested .replace method and it look to fix this.
> 
> Edit: Unsure what's happening up there, but this is a gif pfp, no weird lines?
> https://cdn.discordapp.com/attachments/730127882021503046/815647787009703956/image.gif

The issue is about animated avatars (with Nitro users), the filter can not be used on their avatar. We need to convert their avatar by an simple .png avatar (not animated) with `user.displayAvatarURL({dynamic: false})` to use the filter and to not have these weird  lines.

**Comment by asdbee** [comment]
> > This is fixed with a simple .replace('.gif','.png'). I tested .replace method and it look to fix this.
> 
> > 
> 
> > Edit: Unsure what's happening up there, but this is a gif pfp, no weird lines?
> 
> > https://cdn.discordapp.com/attachments/730127882021503046/815647787009703956/image.gif
> 
> 
> 
> The issue is about animated avatars (with Nitro users), the filter can not be used on their avatar. We need to convert their avatar by an simple .png avatar (not animated) with `user.displayAvatarURL({dynamic: false})` to use the filter and to not have these weird  lines.

I'm aware that it is with Nitro users. The gif shown in my comment was **with** a Nitro (gif) profile picture. I had no issue adding effects to it.

**Comment by Mr-KayJayDee** [comment]
The issue I get is for the triggered generation and blink because they are gifs and so we would need to add a gif in a gif and it is hard to know the duration of the gif than decompose this gif, add every frame and generate the image for every frame and then send the file. I think you can imagine the duration of this thing 🤯

**Comment by asdbee** [comment]
What I am saying is, there is **no lines or distortion** on the gifs I'm generating with **gif** effects. 

Take a look at this video. https://streamable.com/usud87

**Comment by Mr-KayJayDee** [comment]
Yeah, maybe that is only some gif that aren't supported i did not took time to test it nor think of it.

**Comment by Mr-KayJayDee** [comment]
Basically it does like discord avatars when you have nitro, it takes the first frame of the gif and use it for the generation.
